### PR TITLE
Fixes #6049 - Default provider [files] section always executed

### LINKED ISSUE #6049: Default provider [files] section always executed
**Jetty version**
10.0.x

**Description**
`java -jar start.jar --add-modules=http,logging-logback` copies `resources/jetty-logging.properties`, but it should not.

Interestingly, 
`java -jar start.jar --add-modules=logging-logback,http` does *not* copy `resources/jetty-logging.properties`.

Since an explicit provider has been configured, the default provider should be out of the picture.

### PR DESCRIPTION
**Pull request by sbordet**
Keeping only enabled modules when processing the modules,
so that default provider modules don't get processed.

Signed-off-by: Simone Bordet <simone.bordet@gmail.com>

### CODE REVIEW

File: jetty-start/src/main/java/org/eclipse/jetty/start/BaseBuilder.java [hunk 1]
```diff
@@ -264,56 +271,65 @@ public boolean accept(Path entry) throws IOException
             StartLog.warn("Use of both %s and %s is deprecated", getBaseHome().toShortForm(startd), getBaseHome().toShortForm(startini));
 
         builder.set(useStartD ? new StartDirBuilder(this) : new StartIniBuilder(this));
-        newlyAdded.stream().map(modules::get).forEach(module ->
-        {
-            String ini = null;
-            try
+        newlyAdded.stream()
```
Comment (gregw) [inline jetty-start/src/main/java/org/eclipse/jetty/start/BaseBuilder.java:277]: This method is a monster with lots of complex conditionals and loops.
I think perhaps now we probably need to split and/or better document each section of this method.

File: jetty-start/src/main/java/org/eclipse/jetty/start/BaseBuilder.java [hunk 2]
```diff
@@ -264,56 +271,65 @@ public boolean accept(Path entry) throws IOException
...
+            .map(modules::get)
+            .filter(Module::isEnabled)
+            .forEach(module ->
             {
-                if (module.isSkipFilesValidation())
-                {
-                    StartLog.debug("Skipping [files] validation on %s", module.getName());
-                }
-                else
+                StartLog.debug("%s.isEnabled() = %b", module.getName(), module.isEnabled());
```
Comment (gregw) [inline jetty-start/src/main/java/org/eclipse/jetty/start/BaseBuilder.java]: don't need to log isEnabled, as it is always true

**Review by gregw — DISMISSED** [review]
I think the change is good, but I also think we need to better describe the algorithm in the build method so we can actually be sure the change is good.

File: jetty-start/src/main/java/org/eclipse/jetty/start/BaseBuilder.java (hunk 1 above)
Comment (joakime) [reply jetty-start/src/main/java/org/eclipse/jetty/start/BaseBuilder.java:277]: But not for this PR, a new PR for that is a better choice.

Comment (gregw) [reply jetty-start/src/main/java/org/eclipse/jetty/start/BaseBuilder.java:277]: Then this PR at least needs more comments to explain how/why it fixes the problem

**Comment by gregw** [comment]
@sbordet what is the status of this?

**Comment by gregw** [comment]
@sbordet I'm going to remove my review, so you can fix my comments and get somebody else to re-review in your timezone rather than wait 24 hours for me.

**Comment by sbordet** [comment]
@gregw @joakime nudge

**Review by joakime — APPROVED** [review]

**Review by gregw — APPROVED** [review]

**Pull request merged**